cli/server.go

```diff
@@ -2157,6 +2157,10 @@ func ConnectToPostgres(ctx context.Context, logger slog.Logger, driver string, d
 
 	err = migrations.Up(sqlDB)
 	if err != nil {
+		// Checks for database schema version mismatch
+		if errors.Is(err, os.ErrNotExist) && strings.Contains(err.Error(), "no migration found for version") {
```

Comment:
`golang-migrate` does not provide a better way to handle these errors, it returns the error message as a string as seen here

https://github.com/golang-migrate/migrate/blob/c378583d782e026f472dff657bfd088bf2510038/migrate.go#L809

Reply:
What about `migrations.CheckLatestVersion`? Would this not accomplish something similar?

Reply:
I like @johnstcn's suggestion of checking the version first before trying to run the migrations over the string matching which could break. 👍

Reply:
```suggestion
		if errors.Is(err, os.ErrNotExist) {
			if cleanErr := migrations.EnsureClean(sqlDB); cleanErr != nil {
				return nil, xerrors.New("Current database schema requires a newer version of Coder!")
			}
		}
```

@mafredri @johnstcn I could do something like this. This check cannot precede `migrations.Up(sqlDB)` cause we want to run the migrations if the executable is the latest and the migrations have not yet been executed.

wdyt?